engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -238,15 +210,15 @@ public ConstraintValidatorManager getConstraintValidatorManager() {
 	 * created for parameter validation, {@code null} otherwise.
 	 */
 	public List<String> getParameterNames() {
-		if ( parameterNameProvider == null ) {
+		if ( executableParameters == null ) {
```

Comment:
well ... honestly it bugs me, I mean not this particular change but Validation context. In particular two things:
- this null check was previously for the name provider as we were not passing it if the context wasn't for the parameters
- and `createConstraintViolation` method, this part in particular https://github.com/marko-bekhta/hibernate-validator/blob/d976b480573871c37c844ba8f44ffa17b268d9b6/engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java#L294 (If we have a `@NotNull` on return value and the value is actually null it will slip this if and fall to the last one. It is not visible in a constraint but still it kind of bugs me...)

So as I'm not sure about it I wanted to ask you - if it would be a good idea to split this context into 3 different ones ?

Reply:
Yes, you're right, the test in `createConstraintViolation()` is buggy.

About separating `ValidationContext` in three contexts, I think it's too big of a change for a minor. We could maybe consider it for 6.1 but I'm not sure it's worth it.

I think what we are missing here is a `ValidationType` enum to define the type of validation currently at stake. Then we could base the conditions on the type and that would be totally safe. It would be a private member of `ValidationContext` so the change would be very local.

WDYT?

Reply:
sounds good. I'll add an enum for now and create a ticket for splitting the context, so we can decide later if it is worth it or not ...

Reply:
Maybe, it's better to talk about "operation" instead of "type".